### PR TITLE
Get vault submission to nextJS page

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

This PR creates a new page `/admin/vault` and a new api endpoint `/api/retrieval` that request through a lambda function submissions that have been stored in the vault for a specific form.  Both the `/admin/vault` page and `/api/retrieval` api are protected through auth.  Designs are obviously to be iterated upon.

| Search for Form 45 | Response 1 | Response 2 |
|:-----:|:----:|:----:|
|Screen Shot 2021-05-19 at 1 39 39 PM| Screen Shot 2021-05-19 at 1 40 14 PM | Screen Shot 2021-05-19 at 1 40 24 PM

### CODE REVIEW

File: pages/api/submit.js [hunk 1]
```diff
@@ -41,11 +41,12 @@ const callLambda = async (form, fields) => {
   const lambdaClient = new LambdaClient({ region: "ca-central-1" });
 
   const command = new InvokeCommand({
-    FunctionName: process.env.SUBMISSION_API ?? "",
+    FunctionName: process.env.SUBMISSION_API ?? "Submission",
     Payload: JSON.stringify({
       form,
       responses: fields,
       submission,
+      vault: true,
```
Comment (brdunfield) [inline pages/api/submit.js:49]: Not related to this PR, but eventually will we want this to be an option? As in, folks could use either email or vault (or any other) submission method we support?

Comment (bryan-robitaille) [reply pages/api/submit.js:49]: Exactly!

Comment (fitore-cds) [reply pages/api/submit.js:49]: Yup, I think I'll add a minimal "feature flag" functionality to our app, where we can turn features of/off via `.env` variables. This is a good candidate, and showing/hiding the "Login" link in the header is another :-)

**Review by brdunfield — APPROVED** [review]
LGTM and worked well when I tested locally. I'm not sure why your tests are failing - doesn't look like it failed on a file you changed.

**Review by fitore-cds — APPROVED** [review]
What a beauty, love seeing those screenshots! Post on the team channel?

File: pages/api/submit.js (hunk 1 above)
Comment (bryan-robitaille) [reply pages/api/submit.js:49]: Just a reminder that env vars add complexity.  For every env var change there are 3-4 files in terraform that need to be changed and redeployed.

Comment (bryan-robitaille) [reply pages/api/submit.js:49]: On Portal we used a package that used a DB table to store flags that were controlled through Admin Page access.
Really help cut down on the number of infrastructure changes needed as new features were added.

**Pull request merged**